Add the null value

yannham opened the pull request:
Some time ago, we agreed upon having `null` in the language, as we aim for easy interoperability with JSON. With the recent addition of multiple export and import formats, the time has come, especially since #282: if one tries to import a JSON or a YAML file with null values, this currently raises an error.

This PR adds the null value to the language. `null` is typechecked as `Dyn`. Serialization prevalidation ensures that formats that don't support null (such as TOML, for example) raise an appropriate error at export.

edolstra commented:
I'm getting a compilation error if I try to merge this locally:
```
error[E0061]: this function takes 2 arguments but 1 argument was supplied
    --> src/operation.rs:1398:17
     |
1398 |                 serialize::validate(&rt2)?;
     |                 ^^^^^^^^^^^^^^^^^^^ ---- supplied 1 argument
     |                 |
     |                 expected 2 arguments
     |
note: function defined here
    --> src/serialize.rs:70:8
     |
70   | pub fn validate(t: &RichTerm, format: ExportFormat) -> Result<(), SerializationError> {
     |        ^^^^^^^^ ------------  --------------------
```

BTW maybe we should configure the github action to try to test the PR against master if it's not already doing that.

yannham commented:
I rebased and fixed the compilation error.

> BTW maybe we should configure the github action to try to test the PR against master if it's not already doing that.

Indeed

yannham commented:
> BTW maybe we should configure the github action to try to test the PR against master if it's not already doing that.

After looking at this, GH actions should have done that. We have a github action on `pull_request` event, which includes by default `opening` and `synchronizing` (ie, pushing to the branch) events. According to [this discussion](https://github.com/actions/checkout/issues/15#issuecomment-524093065), the default behavior on `pull_request` is to checkout the merge commit. I don't understand why the build/test action doesn't even run on this PR. Bad timing between #274 and disabling the old CI?

regnat commented:
> I don't understand why the build/test action doesn't even run on this PR

There was an issue with GH actions this morning which apparently prevented some actions from being triggered (https://www.githubstatus.com/incidents/xn0sd2x4nd7f). Quite possibly that's the reason here and just re-pushing anything would resolve it
